resources/compress.cuh

```diff
@@ -107,6 +107,9 @@ KERNEL void compress_large(const uint8_t *in, uint32_t *out, uint64_t *scratch,
        Based on "Single-pass Parallel Prefix Scan with Decoupled Look-back"
        by Duane Merrill and Michael Garland */
     while (true) {
+        // Avoid re-ordering of loads and stores under the relaxed memory model.
+        __threadfence_block();
```

Comment:
I have not tested the performance impact of this change, and I'm not 100% certain that:
- This is exactly the kind of fence we need.
- It is placed in the right location.

Same for the other fence in `block_prefix_reduce()`.

Reply:
→ replaced with `volatile` loads